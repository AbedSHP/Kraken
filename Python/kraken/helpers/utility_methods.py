

def logHierarchy(kObject):
    """Traverses the given Kraken hierarchy and logs the names of all the objects.

    Return:
    None

    """

    print kObject.getFullName()
    for i in xrange(kObject.getNumChildren()):
        child = kObject.getChildByIndex(i)
        logHierarchy(child)



from kraken.core.maths.math_object import MathObject
from kraken.core.maths import decodeValue

def __convertFromJSON(jsonData):

    if type(jsonData) is list:
        newList = []
        for item in jsonData:
            newList.append(__convertFromJSON(item))
        return newList
    elif type(jsonData) is dict:
        if '__mathObjectClass__' in jsonData.keys():
            return decodeValue(jsonData)
        for key, value in jsonData.iteritems():
            jsonData[key] = __convertFromJSON(value)
    return jsonData

def prepareToLoad(jsonData):

    return __convertFromJSON(jsonData)

def __convertToJSON(jsonData):

    if isinstance(jsonData, MathObject):
        return jsonData.jsonEncode()
    elif type(jsonData) is list:
        newList = []
        for item in jsonData:
            newList.append(__convertToJSON(item))
        return newList
    elif type(jsonData) is dict:
        for key, value in jsonData.iteritems():
            jsonData[key] = __convertToJSON(value)
    return jsonData

def prepareToSave(jsonData):
    return __convertToJSON(jsonData)