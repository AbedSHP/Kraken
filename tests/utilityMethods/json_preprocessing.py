import json

from kraken.examples.bob_guide_data import bob_guide_data
from kraken.helpers.utility_methods import prepareToSave, prepareToLoad


pureJSON = prepareToSave(bob_guide_data)
print json.dumps(pureJSON, indent=2)

bob_guide_data = prepareToLoad(pureJSON)


