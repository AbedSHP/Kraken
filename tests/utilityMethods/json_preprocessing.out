{
  "connections": [
    {
      "source": "spine.spineEnd", 
      "_comment": "Neck to Spine", 
      "target": "neck.neckBase"
    }, 
    {
      "source": "neck.neckEnd", 
      "_comment": "Head to Neck", 
      "target": "head.headBase"
    }, 
    {
      "source": "spine.spineEnd", 
      "_comment": "LClavicle to Spine", 
      "target": "L_Clavicle.spineEnd"
    }, 
    {
      "source": "L_Clavicle.clavicleEnd", 
      "_comment": "LArm to LClavicle", 
      "target": "L_Arm.clavicleEnd"
    }, 
    {
      "source": "R_Clavicle.clavicleEnd", 
      "_comment": "RArm to RClavicle", 
      "target": "R_Arm.clavicleEnd"
    }, 
    {
      "source": "spine.spineEnd", 
      "_comment": "RClavicle to Spine", 
      "target": "R_Clavicle.spineEnd"
    }, 
    {
      "source": "L_Arm.armEndXfo", 
      "_comment": "LHand To LArm Connections. armEndXfo", 
      "target": "L_Hand.armEndXfo"
    }, 
    {
      "source": "R_Arm.armEndXfo", 
      "_comment": "RHand To RArm Connections. armEndXfo", 
      "target": "R_Hand.armEndXfo"
    }, 
    {
      "source": "spine.spineBase", 
      "_comment": "LLeg To Pelvis Connections", 
      "target": "L_Leg.pelvisInput"
    }, 
    {
      "source": "spine.spineBase", 
      "_comment": "RLeg To Pelvis Connections", 
      "target": "R_Leg.pelvisInput"
    }, 
    {
      "source": "L_Leg.legEndXfo", 
      "_comment": "LFoot To LLeg Connections: legEndXfo", 
      "target": "L_Foot.legEndXfo"
    }, 
    {
      "source": "R_Leg.legEndXfo", 
      "_comment": "RFoot To RLeg Connections: legEndXfo", 
      "target": "R_Foot.legEndXfo"
    }
  ], 
  "components": [
    {
      "spine03Position": {
        "y": 12.449600219726562, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.36489999294281006
      }, 
      "spine01Position": {
        "y": 11.135100364685059, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.13819999992847443
      }, 
      "numDeformers": 6, 
      "cogPosition": {
        "y": 11.135100364685059, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.13819999992847443
      }, 
      "spine02Position": {
        "y": 11.801300048828125, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.19949999451637268
      }, 
      "class": "kraken.examples.spine_component.SpineComponentGuide", 
      "spine04Position": {
        "y": 13.10509967803955, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.4821000099182129
      }
    }, 
    {
      "neckPosition": {
        "y": 16.557199478149414, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.6915000081062317
      }, 
      "class": "kraken.examples.neck_component.NeckComponentGuide", 
      "neckEndPosition": {
        "y": 17.47559928894043, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.42100000381469727
      }
    }, 
    {
      "jawPosition": {
        "y": 17.613000869750977, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.27309998869895935
      }, 
      "eyeLeftPosition": {
        "y": 18.087799072265625, 
        "x": 0.349700003862381, 
        "__mathObjectClass__": "Vec3", 
        "z": 0.6087999939918518
      }, 
      "headPosition": {
        "y": 17.47559928894043, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.42100000381469727
      }, 
      "headEndPosition": {
        "y": 19.5, 
        "x": 0.0, 
        "__mathObjectClass__": "Vec3", 
        "z": -0.42100000381469727
      }, 
      "class": "kraken.examples.head_component.HeadComponentGuide", 
      "eyeRightPosition": {
        "y": 18.087799072265625, 
        "x": -0.349700003862381, 
        "__mathObjectClass__": "Vec3", 
        "z": 0.6087999939918518
      }
    }, 
    {
      "name": "L_Clavicle", 
      "clavicleUpVXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 1.0, 
          "x": 0.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 0.0
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "clavicleEndXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 15.295000076293945, 
          "x": 2.2699999809265137, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.753000020980835
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "location": "L", 
      "class": "kraken.examples.clavicle_component.ClavicleComponentGuide", 
      "clavicleXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 15.402999877929688, 
          "x": 0.13220000267028809, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.5723000168800354
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }
    }, 
    {
      "name": "R_Clavicle", 
      "clavicleUpVXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 1.0, 
          "x": 0.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 0.0
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "clavicleEndXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 15.295000076293945, 
          "x": -2.2699999809265137, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.753000020980835
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "location": "R", 
      "class": "kraken.examples.clavicle_component.ClavicleComponentGuide", 
      "clavicleXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 15.402999877929688, 
          "x": -0.13220000267028809, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.5723000168800354
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }
    }, 
    {
      "forearmFKCtrlSize": 1.5, 
      "wristXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 12.281900405883789, 
          "x": 7.188600063323975, 
          "__mathObjectClass__": "Vec3", 
          "z": 0.49059998989105225
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "name": "L_Arm", 
      "bicepXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 15.295000076293945, 
          "x": 2.2699999809265137, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.753000020980835
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "bicepFKCtrlSize": 1.75, 
      "forearmXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 13.5600004196167, 
          "x": 5.039000034332275, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.859000027179718
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "class": "kraken.examples.arm_component.ArmComponentGuide", 
      "location": "L"
    }, 
    {
      "forearmFKCtrlSize": 1.5, 
      "wristXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 12.281900405883789, 
          "x": -7.188600063323975, 
          "__mathObjectClass__": "Vec3", 
          "z": 0.49059998989105225
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "name": "R_Arm", 
      "bicepXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 15.295000076293945, 
          "x": -2.2699999809265137, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.753000020980835
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "bicepFKCtrlSize": 1.75, 
      "forearmXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 13.5600004196167, 
          "x": -5.039000034332275, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.859000027179718
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "class": "kraken.examples.arm_component.ArmComponentGuide", 
      "location": "R"
    }, 
    {
      "location": "L", 
      "class": "kraken.examples.hand_component.HandComponentGuide", 
      "handXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 12.281900405883789, 
          "x": 7.188600063323975, 
          "__mathObjectClass__": "Vec3", 
          "z": 0.49059998989105225
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 0.9330999851226807, 
          "v": {
            "y": -0.23010000586509705, 
            "x": -0.08649999648332596, 
            "__mathObjectClass__": "Vec3", 
            "z": -0.2623000144958496
          }
        }
      }, 
      "name": "L_Hand"
    }, 
    {
      "location": "R", 
      "class": "kraken.examples.hand_component.HandComponentGuide", 
      "handXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 12.281900405883789, 
          "x": -7.188600063323975, 
          "__mathObjectClass__": "Vec3", 
          "z": 0.49059998989105225
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 0.2623000144958496, 
          "v": {
            "y": -0.08649999648332596, 
            "x": -0.23010000586509705, 
            "__mathObjectClass__": "Vec3", 
            "z": -0.9330999851226807
          }
        }
      }, 
      "name": "R_Hand"
    }, 
    {
      "name": "L_Leg", 
      "ankleXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 1.1516000032424927, 
          "x": 1.840999960899353, 
          "__mathObjectClass__": "Vec3", 
          "z": -1.2369999885559082
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "kneeXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 5.441800117492676, 
          "x": 1.448799967765808, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.5347999930381775
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "location": "L", 
      "class": "kraken.examples.leg_component.LegComponentGuide", 
      "femurXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 9.769000053405762, 
          "x": 0.9811000227928162, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.45719999074935913
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }
    }, 
    {
      "location": "L", 
      "class": "kraken.examples.foot_component.FootComponentGuide", 
      "footXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 1.1516000032424927, 
          "x": 1.840999960899353, 
          "__mathObjectClass__": "Vec3", 
          "z": -1.2369999885559082
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 0.4189999997615814, 
          "v": {
            "y": -0.5695000290870667, 
            "x": 0.6377000212669373, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.3052999973297119
          }
        }
      }, 
      "name": "L_Foot"
    }, 
    {
      "name": "R_Leg", 
      "ankleXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 1.1516000032424927, 
          "x": -1.840999960899353, 
          "__mathObjectClass__": "Vec3", 
          "z": -1.2369999885559082
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "kneeXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 5.441800117492676, 
          "x": -1.448799967765808, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.5347999930381775
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }, 
      "location": "R", 
      "class": "kraken.examples.leg_component.LegComponentGuide", 
      "femurXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 9.769000053405762, 
          "x": -0.9811000227928162, 
          "__mathObjectClass__": "Vec3", 
          "z": -0.45719999074935913
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 1.0, 
          "v": {
            "y": 0.0, 
            "x": 0.0, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.0
          }
        }
      }
    }, 
    {
      "location": "R", 
      "class": "kraken.examples.foot_component.FootComponentGuide", 
      "footXfo": {
        "sc": {
          "y": 1.0, 
          "x": 1.0, 
          "__mathObjectClass__": "Vec3", 
          "z": 1.0
        }, 
        "__mathObjectClass__": "Xfo", 
        "tr": {
          "y": 1.1516000032424927, 
          "x": -1.840999960899353, 
          "__mathObjectClass__": "Vec3", 
          "z": -1.2369999885559082
        }, 
        "ori": {
          "__mathObjectClass__": "Quat", 
          "w": 0.3052999973297119, 
          "v": {
            "y": -0.6377000212669373, 
            "x": 0.5695000290870667, 
            "__mathObjectClass__": "Vec3", 
            "z": 0.4189999997615814
          }
        }
      }, 
      "name": "R_Foot"
    }
  ]
}
True
